google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1113,4 +1114,289 @@ public Bucket setDefaultKmsKey(String bucketName, String kmsKeyName) throws Stor
+    System.out.println("Metageneration: " + blob.getMetageneration());
+    System.out.println("Name: " + blob.getName());
+    System.out.println("Size: " + blob.getSize());
+    System.out.println("StorageClass: " + blob.getStorageClass());
+    System.out.println("TimeCreated: " + new Date(blob.getCreateTime()));
+    System.out.println("Last Metadata Update: " + new Date(blob.getUpdateTime()));
+    System.out.println(
+        "temporaryHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    System.out.println(
+        "eventBasedHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    if (blob.getRetentionExpirationTime() != null) {
+      System.out.println("retentionExpirationTime: " + new Date(blob.getRetentionExpirationTime()));
+    }
+    if (blob.getMetadata() != null) {
+      System.out.println("User metadata:");
+      for (Map.Entry<String, String> userMetadata : blob.getMetadata().entrySet()) {
+        System.out.println(userMetadata.getKey() + " " + userMetadata.getValue());
+      }
+    }
+    // [END storage_get_metadata]
+  }
+
+  /** Example of setting a retention policy on a bucket */
+  public Bucket setRetentionPolicy(String bucketName, Long retentionPeriod)
+      throws StorageException {
+    // [START storage_set_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    // The retetion period for objects in bucket
+    // 1 hour in seconds
+    // Long retentionPeriod = 3600L;
+
+    Bucket bucket =
+        storage.update(
+            BucketInfo.newBuilder(bucketName).setRetentionPeriod(retentionPeriod).build());
+
+    System.out.println(
+        "Retention period for " + bucketName + " is now " + bucket.getRetentionPeriod());
+    // [END storage_set_retention_policy]
+    return bucket;
+  }
+
+  /** Example of removing a retention policy on a bucket */
+  public Bucket removeRetentionPolicy(String bucketName) throws StorageException {
+    // [START storage_remove_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    Bucket bucket = storage.get(bucketName, BucketGetOption.fields(BucketField.RETENTION_POLICY));
+    if (bucket.retentionPolicyIsLocked() != null && bucket.retentionPolicyIsLocked()) {
+      System.out.println("Unable to remove retention period as retention policy is locked.");
+      return bucket;
+    }
+
+    Bucket updated_bucket = bucket.toBuilder().setRetentionPeriod(null).build().update();
```

Comment:
Are buckets immutable? Do you need to create a new builder instead of just updating the original bucket? What's the difference between `bucket.update` vs `storage.update(bucket)`?

Reply:
That's correct, buckets are immutable. And `bucket.update()` calls `storage.update()` under the hood.